### PR TITLE
`Get`: Fix handling of paths with number template literal

### LINKED ISSUE #967: `Get<T, Paths>` resolves to `unknown` if `Paths` contains a template literal type
### Bug description

The code from the repro:

```ts
import type { Get } from 'type-fest'

interface Foo {
  arr: string[]
}

type FooPaths = `arr.${number}`

// Should be 'string | undefined'
// Is actually 'unknown'
type FooValues = Get<Foo, FooPaths>
```

FooValues resolves correctly for any finite intersection of the form 'arr' | `arr.1` | `arr.2` | …

### Repro

https://www.typescriptlang.org/play/?#code/JYWwDg9gTgLgBDAnmApnA3nA4i+BfOAMyghDgHIlUBaQlAZxnIFgAoN4AOxhSkIEMAxmgBiECBjZw4-KFABccRlC4BzANoBdNnjZsqo8QAV+MABb04AXjgADWVAB0AEnScAriABGvPLb2sAPSBcADKZhDuADYAJnA+FMpqcAA+cO6cMSiEXCgxLEEhAJKWQjDu-FFRiBQZANacEADunAUGcGIQAGqV7gzW2LgAPJ0ANB3GphYAfEA

### PR DESCRIPTION
**Pull request by sindresorhus**
Fixes #967

### CODE REVIEW

File: test-d/get.ts [hunk 1]
```diff
@@ -139,3 +139,13 @@ expectTypeOf<Get<{a: readonly []}, 'a[0]'>>().toEqualTypeOf<unknown>();
...
+
+// eslint-disable-next-line no-lone-blocks
+{
+	type Foo = {
+		array: string[];
+	};
+
+	type FooPaths = `array.${number}`;
+	expectTypeOf<Get<Foo, FooPaths>>().toEqualTypeOf<string | undefined>();
+}
```
Comment (Emiyaaaaa) [inline test-d/get.ts:151]: Add Test:
```
type FooPaths2 = `array.1`;
expectTypeOf<Get<Foo, FooPaths2>>().toEqualTypeOf<string | undefined>();
```

**Review by Emiyaaaaa — CHANGES_REQUESTED** [review]

File: test-d/get.ts (hunk 1 above)
Comment (sindresorhus) [reply test-d/get.ts:154]: Added

**Review by Aliqag — APPROVED** [review]

**Review by Sexypella — APPROVED** [review]

**Comment by KelvinNelly** [comment]
https://github.com/KelvinNelly

**Comment by KelvinNelly** [comment]
https://t.me/+DFz0KZdBu9U0NWM0
Join our telegram group.

**Pull request merged**